narwhals/_sql/expr_dt.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+from typing import Any, Generic
+
+from narwhals._compliant import LazyExprNamespace
+from narwhals._compliant.any_namespace import DateTimeNamespace
+from narwhals._sql.typing import SQLExprT
+
+
+class SQLExprDateTimeNamesSpace(
+    LazyExprNamespace[SQLExprT], DateTimeNamespace[SQLExprT], Generic[SQLExprT]
+):
+    def _function(self, name: str, *args: Any) -> SQLExprT:
+        return self.compliant._function(name, *args)  # type: ignore[no-any-return]
+
+    def year(self) -> SQLExprT:
+        return self.compliant._with_elementwise(lambda expr: self._function("year", expr))
+
+    def month(self) -> SQLExprT:
+        return self.compliant._with_elementwise(
+            lambda expr: self._function("month", expr)
+        )
+
+    def day(self) -> SQLExprT:
+        return self.compliant._with_elementwise(lambda expr: self._function("day", expr))
```

Comment:
as another follow-up, if you fancy it, maybe we could have an extra
```python
def _simple_function(self, name: str) -> SQLExprT:
    return self.compliant._with_elementwise(lambda expr: self._function(name, expr))
```
and then re-use that here
```python
    def day(self) -> SQLExprT:
        return self._simple_function('day')
```
(and for others)